examples/bidspath.py

```diff
@@ -0,0 +1,159 @@
+# retrieve a `pathlib.Path` object of this path:
+
+pathlib_path = bids_path.fpath
+pathlib_path
+
+###############################################################################
+# Let's have a closer look at the components of our ``BIDSPath`` again.
+
+bids_path
+
+###############################################################################
+# The most interesting thing here is probably the **basename**. It's what
+# MNE-BIDS uses to name individual files. The basename consists of a set of
+# so-called **entities**, which are concatenated using underscores. You can
+# access it directly:
+
+bids_path.basename
+
+###############################################################################
+# The two entities you can see here are the ``subject`` entity (``sub``) and
+# the ``session`` entity (``ses``). Each entity name also has a value; for
+# ``sub``, this is ``123``, and for ``ses``, it is ``test`` in our example.
+# Entity names (or "keys") and values are separated via hyphens.
+# BIDS knows a much larger number of entities, and MNE-BIDS allows you to make
+# use of them. To get a list of all supported entities, use:
+
+bids_path.entities
+
+###############################################################################
+# As you can see, most entity keys are set to ``None``, which is the default
+# and implies that no value has been set. Let us add a ``run`` entity, and
+# remove the ``session``:
+
+run = '01'
+session = None
+bids_path.update(run=run, session=session)
+bids_path
+
+###############################################################################
+# As you can see, the ``basename`` has been updated. In fact, the entire
+# **path** has been updated, and the ``ses-test`` folder has been dropped from
+# the path:
+
+print(bids_path.fpath)
+
+###############################################################################
+# Awesome! We're almost done! Two important things are still missing, though:
+# the so-called **suffix** and the filename **extension**. Sometimes these
+# terms are used interchangably, but in BIDS, they have a very specific
+# and different meaning!
+#
+# The **suffix** is the last part of a BIDS filename before the extension. It
+# is the same as the datatype for MEG, EEG, and iEEG recordings (i.e.
+# ``meg``, ``eeg``, and ``ieeg``, respectively) and ``T1w`` for T1-weighted
+# MRI scans. But the suffix is also used to create the names of sidecar files
+# like ``*_events.tsv``.
+#
+# Which brings us directly to the **extension**: the very last part of a
+# filename. In MNE-BIDS, the extension contains a leading period, e.g.
+# ``.tsv``.
+# Let's put our new knowledge to use!
+
+bids_path.update(suffix='meg', extension='fif')
+print(bids_path.fpath)
+bids_path
+
+###############################################################################
+# By default, most MNE-BIDS functions will try to infer to correct
+# suffix and extension for your data, and you don't need to specify them
+# manually.
```

Comment:
isn't there some functionality to query as well? cc @adam2392 should perhaps review this example too

Reply:
Yes we added `BIDSPath.match` for example, is that what you mean?

Reply:
yes!

Reply:
The example overall looks pretty good to me. It makes sense to merge the 01-example based on what I've seen there. You all critiqued thoroughly already :0